### PR TITLE
Add Topological Data Analysis

### PR DESCRIPTION
**Pull request by FatemehTarashi**
This's #1703 closed pull request.

[topological data analysis](https://github.com/FatemehTarashi/awesome-TDA)
(TDA) is an approach to the analysis of datasets using techniques from topology. 
This list is handy for those who want to know more about TDA's theory and Tools in both mathematics and computer science.

PRs Reviewed: 
#1701 
#1684 
#1792

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.
- You have read and understood the [instructions for creating a list](create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=master&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name or text to the readme. GitHub already shows the license name at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- Has [contribution guidelines](awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

### CODE REVIEW

**Comment by FatemehTarashi** [comment]
unicorn 🦄

**Comment by sindresorhus** [comment]
Please read through the guidelines again. There are many things that are not followed.

**Comment by FatemehTarashi** [comment]
@sindresorhusI fixed it. Now, awesome-lint hasn't any errors.

**Comment by biesiadamich** [comment]
LGTM - no linter errors. Only free suggestions - free lines (redundant 'enters', but looks also OK). And License info - referring to 'no licence info necessary into readme'.

**Comment by FatemehTarashi** [comment]
> LGTM - no linter errors. Only free suggestions - free lines (redundant 'enters', but looks also OK). And License info - referring to 'no licence info necessary into readme'.

Thank you, but I do not agree. free lines don't look different in view, but it helps me. also I think it's better for someone who doesn't want to contribution to see the license without opening a new file.

File: readme.md [hunk 1]
```diff
@@ -354,6 +354,7 @@
 - [Search Engine Optimization](https://github.com/marcobiedermann/search-engine-optimization#readme)
 - [Competitive Programming](https://github.com/lnishan/awesome-competitive-programming#readme)
 - [Math](https://github.com/rossant/awesome-math#readme)
+	- [Topological Data Analysis](https://github.com/FatemehTarashi/awesome-TDA)
```
Comment (mourarthur) [inline readme.md:357]: > Your entry here should include a short description about the project/theme of the list. It should not describe the list itself. The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.

You could go with _"In applied mathematics, TDA is an approach to the analysis of datasets using techniques from topology."_ (which is already in the description in your readme).

**Comment by FatemehTarashi** [comment]
unicorn

**Comment by Sid2697** [comment]
@FatemehTarashi As mentioned in the guidelines,
> Do not add the license name or text to the readme. GitHub already shows the license name at the top of the repo.

please remove the license information from the bottom of your README.

As per the guidelines, please run `awesome-lint` on your repository. [Here](https://github.com/sindresorhus/awesome-lint) is the link to awesome-lint.

> Run awesome-lint on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please report it.

Another point from the guideline which is not followed is, 
> The list item should start with a capital letter and end with proper punctuation.

**Comment by margaretmz** [comment]
An addition to the previous review comments: 
The default branch should be named `main`, not `master`.

**Comment by FatemehTarashi** [comment]
@Sid2697 Thanks for licence part. I run awesome-lint before and now and my repository doesn't have any error.
@margaretmz It does not matter if my repository is on main or master. It is important that the awesome repository forked is on the main.

**Comment by margaretmz** [comment]
> @margaretmz It does not matter if my repository is on main or master. It is important that the awesome repository forked is on the main.

Thanks for your response @FatemehTarashi. I  was interpreting the guideline of "The default branch should be named `main`, not `master`." as your awesome-list repo should have `main` as the default branch. It's under the **Requirements for your Awesome list** section.

**Comment by sindresorhus** [comment]
There are still many things that are not followed:

---

> Includes a succinct description of the project/theme at the top of the readme. (Example)


---

> Entries have a description, unless the title is descriptive enough by itself. It rarely is though.

---

And more.

**Comment by sindresorhus** [comment]
I'm going to close this for now as the list is not close to being added and there's no progress. I would recommend reading through the guidelines more closely, work more on the list, and try submitting again sometime in the future (after at least 3 months).

**Pull request closed without merging**